src/lib/utils.ts

```diff
@@ -77,9 +73,33 @@ export function parseNid(
   return undefined;
 }
 
+export function parseRepositoryId(
+  rid: string,
+): { prefix: string; pubkey: string } | undefined {
+  const match = /^(rad:)?(z[a-zA-Z0-9]+)$/.exec(rid);
+  if (match) {
+    const hex = bs58.decode(match[2].substring(1));
+    if (hex.byteLength !== 20) {
+      return undefined;
+    }
+
+    return { prefix: match[1] || "rad:", pubkey: match[2] };
+  }
+
+  return undefined;
+}
+
+export function isNodeId(input: string): boolean {
+  return !!parseNodeId(input);
```

Comment:
Instead of a double negation shouldn't we be able to do here
```suggestion
  return Boolean(parseNodeId(input));
```

Reply:
5696b913e4687afb0a54aa4eaa4e159769369204